p3-weather-journal-app/website/app.js

```diff
@@ -3,3 +3,111 @@
 // Create a new date instance dynamically with JS
 let d = new Date();
 let newDate = d.getMonth() + "." + d.getDate() + "." + d.getFullYear();
+
+// Function to get Data from API
+const getCityCords = async (cityURL, city, API_KEY) => {
+  //get city coordinates: https://openweathermap.org/api/geocoding-api
+  const url = `${cityURL}${city}&limit=1&${API_KEY}`;
+  const response = await fetch(url); //fetch return promise
+  let geodata = await response.json();
+  //response format as above doc
+  console.log(geodata[0].lat, geodata[0].lon);
+  return geodata;
+};
+
+const getWeatherData = async (baseURL, city, API_KEY) => {
+  let geodata = await getCityCords(cityURL, city, API_KEY);
+  //weather api: https://openweathermap.org/api/one-call-3
+  const url = `${baseURL}lat=${geodata[0].lat}&lon=${geodata[0].lon}&exclude=minutely,hourly,daily,alerts&${API_KEY}`;
+  //const response = await fetch(url);
+  const response = await fetch(url);
+  let data = await response.json();
+  return data;
+};
+//getWeatherData(baseURL,'dubai',API_KEY)
+
+document.getElementById("generate").addEventListener("click", startGenerate);
+
+function startGenerate() {
+  const city = document.getElementById("city").value;
+  const feelings = document.getElementById("feelings").value;
+  getWeatherData(baseURL, city, API_KEY)
+    .then(function (data) {
+      // Add data
+      console.log("AllData from api: ", data);
+      postDataApi("addWeatherData", {
+        temperature: data.current.temp,
+        date: convertDate(data.current.dt),
+        userResponse: feelings,
+      });
+    })
+    .then(() => updateUI());
+}
+
+//Function to POST data
+const postDataApi = async (url = "", data = {}) => {
+  console.log("post weather data: ", data);
+  const response = await fetch(url, {
+    method: "POST",
+    credentials: "same-origin",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(data), // body data type must match "Content-Type" header
+  });
+
+  try {
+    const newData = await response.json();
+    console.log("post res: ", newData);
+  } catch (error) {
+    console.log("error", error);
```

Comment:
use console.warn in catch, and do console.warn(error.message)

Reply:
done in: [20ecb96](https://github.com/leemoispace/Udacity-Front-End-WebDeveloper/pull/7/commits/20ecb96ba6c5ac532cbc0ea394143deb379eb132)